python/ray/data/dataset.py

```diff
@@ -127,6 +128,8 @@ def __init__(
         read methods to construct a dataset.
         """
         assert isinstance(plan, ExecutionPlan)
+        usage_lib.record_library_usage("dataset")
```

Comment:
For native libraries, should we create a constant for the lib name?

Reply:
I think it's ok. Users can record anything they want: not just libraries, but also particular library features.

Reply:
I think it's ok. Users can record anything they want: not just libraries, but also particular library features.